lib-multisrc/gigaviewer/src/eu/kanade/tachiyomi/multisrc/gigaviewer/GigaViewerDto.kt

```diff
@@ -24,3 +31,56 @@ data class GigaViewerPage(
     val type: String = "",
     val width: Int = 0,
 )
+
+@Serializable
+class GigaViewerPaginationReadableProduct(
+    val description: String?,
+    val display_open_at: String?,
+    val event_free_open_limit: String?,
+    val free_term_start_at: String?,
+    val open_limit: String?,
+    val readable_product_id: String = "",
+    val status: GigaViewerPaginationReadableProductStatus?,
+    val thumbnail_alt: String = "",
+    val thumbnail_uri: String = "",
+    val title: String = "",
+    val viewer_uri: String = "",
+) {
+    fun toSChapter(chapterListMode: Int, publisher: String) = SChapter.create().apply {
+        name = title
+        if (chapterListMode == CHAPTER_LIST_PAID && status?.label != IS_FREE) {
+            name = YEN_BANKNOTE + name
+        } else if (chapterListMode == CHAPTER_LIST_LOCKED && status?.label == UNPUBLISHED) {
+            name = LOCK + name
+        }
+        date_upload = display_open_at?.toDate() ?: 0L
+        scanlator = publisher
+        url = "/episode/$readable_product_id"
+    }
+
+    companion object {
+        // chapter status labels
+        private const val IS_FREE = "is_free"
+        private const val IS_RENTABLE = "is_rentable"
+        private const val IS_PURCHASABLE = "is_purchasable"
+        private const val UNPUBLISHED = "unpublished"
```

Comment:
Inline these constants. If you want to note down the possible values, add a comment beside the field.